src/scripts/argo_add_repository.sh

```
@@ -0,0 +1,73 @@
+if ! command -v argocd >/dev/null 2>&1; then
+  echo -e "${RED}Error: argocd CLI is not installed or not in PATH.${NC}"
+  exit 2
+fi
+
+# Validate that the ARGO_CLI_COMMON_SCRIPT variable is set
+if [[ -z "${ARGO_CLI_COMMON_SCRIPT:-}" ]]; then
+  echo -e "${RED}Error: ARGO_CLI_COMMON_SCRIPT is empty${NC}"
+  exit 2
+fi
+
+source <(echo "$ARGO_CLI_COMMON_SCRIPT")
```

Shellcheck (0.10.0)

[warning] ShellCheck can't follow non-constant source. Use a directive to specify location.

(SC1090)